Mathlib/NumberTheory/MaricaSchoenheim.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Combinatorics.SetFamily.FourFunctions
+import Mathlib.Data.Nat.Squarefree
+
+/-!
+# The Marica-Schönheim special case of Graham's conjecture
+
+Graham's conjecture states that if $0 < a_1 < \dots a_n$ are integers, then
+$\max_{i, j} \frac{a_i}{\gcd(a_i, a_j)} \ge n$. This file proves the conjecture when the $a_i$ are
+squarefree as a corollary of the Marica-Schönheim inequality.
```

Comment:
You are calling the corollary itself `marica_schoenheim`...

Reply:
Yes, as in "the Marica-Schönheim special case of Graham's conjecture". Do you have suggestions for a better name?

Reply:
`GrahamConjectureFor.of_squarefree`? Then you should define something like
```
def GrahamConjectureFor (n : ℕ) (f : Fin n ↪o ℕ) : Prop :=
  n ≠ 0 → ∃ i j, (f i).gcd (f j) * n ≤ f i
```

Reply:
Hmm... maybe, but Graham's conjecture has been proved in the general case. Proof is hard though. Here I'm just trying to demonstrate the use of the Marica-Schönheim inequality.

Reply:
Then this `def` would be a good place to explain that the general proof is available but has not been formalized yet.

Reply:
What about now?